backend/inspirehep/matcher/api.py

```diff
@@ -0,0 +1,117 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2020 CERN.
+#
+# inspirehep is free software; you can redistribute it and/or modify it under
+# the terms of the MIT License; see LICENSE file for more details.
+
+import requests
+from flask import current_app
+from inspire_dojson.utils import get_recid_from_ref, get_record_ref
+from inspire_matcher import match
+from inspire_utils.dedupers import dedupe_list
+from inspire_utils.record import get_value
+
+from .utils import grobid_to_reference
+
+
+def get_reference_from_grobid(query):
+    data = {"citations": query}
+    url = current_app.config["GROBID_URL"]
+    result_xml = requests.post(url, data=data).content
+    return grobid_to_reference(result_xml)
+
+
+def _add_match_to_reference(reference, matched_recid, es_index):
+    """Modifies a reference to include its record id."""
+    if es_index == "records-data":
+        reference["record"] = get_record_ref(matched_recid, "data")
+    elif es_index == "records-hep":
+        reference["record"] = get_record_ref(matched_recid, "literature")
+
+
+def match_reference_with_config(reference, config, previous_matched_recid=None):
+    """Match a reference using inspire-matcher given the config.
+    Args:
+        reference (dict): the metadata of the reference.
+        config (dict): the list of inspire-matcher configurations for queries.
+        previous_matched_recid (int): the record id of the last matched
+            reference from the list of references.
+    Returns:
+        dict: the matched reference.
+    """
+    # XXX: avoid this type casting.
+    try:
```

Comment:
Shouldn't we fix code which expects year to be string than cast this like that and then recasting it back to int at the end? This is really ugly.

Reply:
I changed it a bit